### PR TITLE
Update store account (local state) poc

### PR DESCRIPTION
**Pull request by ratik21**
+ rename `StoreAccountImpl` -> `StoreAccount`, `StoreAccount -> StoreAccountI`
+ update `appsLocalState` from `array` to `Map`

Origin: [https://github.com/scale-it/algorand-builder/pull/177#discussion_r548490788](https://github.com/scale-it/algorand-builder/pull/177#discussion_r548490788)

### CODE REVIEW

File: examples/crowdfunding/test/crowdfundingTest.js [hunk 1]
```diff
@@ -127,8 +127,8 @@ describe('Crowdfunding Tests', function () {
     await runtime.optInToApp(donor.address, applicationId, {}, {}, program);
 
     syncAccounts();
-    assert.isDefined(creator.appsLocalState.find(app => app.id === applicationId));
-    assert.isDefined(donor.appsLocalState.find(app => app.id === applicationId));
+    assert.isDefined(creator.appsLocalState.get(applicationId));
+    assert.isDefined(donor.appsLocalState.get(applicationId));
```
Comment (robert-zaremba) [inline examples/crowdfunding/test/crowdfundingTest.js:131]: :+1:

File: packages/algorand-js/src/runtime/account.ts [hunk 2]
```diff
@@ -13,14 +13,14 @@ import { ERRORS } from "../errors/errors-list";
...
+import { StackElem, StoreAccountI } from "../types";
 
-export class StoreAccountImpl implements StoreAccount {
+export class StoreAccount implements StoreAccountI {
   readonly account: Account;
   readonly address: string;
   assets: AssetHolding[]; // TODO: to be removed
   amount: number;
-  appsLocalState: AppLocalState[]; // TODO: update to map
+  appsLocalState: Map<number, AppLocalState>; // TODO: update to map
```
Comment (robert-zaremba) [inline packages/algorand-js/src/runtime/account.ts:23]: you can remove `TODO` comment.

File: packages/algorand-js/src/runtime/account.ts [hunk 3]
```diff
@@ -75,25 +75,23 @@ export class StoreAccountImpl implements StoreAccount {
...
-        assertValidSchema(l["key-value"], l.schema); // verify if updated schema is valid by config
-        return localState;
+      if (idx === -1) {
+        localApp.push(data); // push new pair if key not found
+      } else {
+        localApp[idx].value = data.value; // update value if key found
       }
+      localState["key-value"] = localApp; // save updated state
+
+      assertValidSchema(localState["key-value"], localState.schema); // verify if updated schema is valid by config
```
Comment (robert-zaremba) [inline packages/algorand-js/src/runtime/account.ts:93]: `"key-value"` is used few times. Please use `const` value for that, instead of strings.

**Review by robert-zaremba — APPROVED** [review]

File: packages/algorand-js/src/runtime/account.ts (hunk 2 above)
Comment (ratik21) [reply packages/algorand-js/src/runtime/account.ts:23]: ok

File: packages/algorand-js/src/runtime/account.ts (hunk 3 above)
Comment (ratik21) [reply packages/algorand-js/src/runtime/account.ts:93]: Okay

**Pull request merged**